ext/spl/spl_directory.c

```diff
@@ -2510,25 +2506,11 @@ PHP_METHOD(SplFileObject, flock)
 
 	CHECK_SPL_FILE_OBJECT_IS_INITIALIZED(intern);
 
-	act = operation & PHP_LOCK_UN;
-	if (act < 1 || act > 3) {
+	if (FAILURE == php_flock_common(intern->u.file.stream, operation, wouldblock, return_value)) {
 		zend_argument_value_error(1, "must be either LOCK_SH, LOCK_EX, or LOCK_UN");
 		RETURN_THROWS();
 	}
-
-	if (wouldblock) {
-		ZEND_TRY_ASSIGN_REF_LONG(wouldblock, 0);
-	}
-
-	/* flock_values contains all possible actions if (operation & PHP_LOCK_NB) we won't block on the lock */
-	act = flock_values[act - 1] | (operation & PHP_LOCK_NB ? LOCK_NB : 0);
-	if (php_stream_lock(intern->u.file.stream, act)) {
-		if (operation && errno == EWOULDBLOCK && wouldblock) {
-			ZEND_TRY_ASSIGN_REF_LONG(wouldblock, 1);
-		}
-		RETURN_FALSE;
-	}
-	RETURN_TRUE;
+	/* return_value is already set by php_parse_flock_arguments */
```

Comment:
Comment doesn't match function name ... and is kinda redundant.

Reply:
I think I would prefer it if you passed in the operation_arg_no and let php_flock_common handle the error case as well. Without SUCCESS/FAILURE return value.